pkg/secrets/info_nix_test.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build secrets && !windows
+// +build secrets,!windows
+
+package secrets
+
+import (
+	"bytes"
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	testConfInfo = []byte(`---
+instances:
+- password: ENC[pass3]
+- password: ENC[pass2]
+`)
+)
+
+func TestGetExecutablePermissionsError(t *testing.T) {
+	secretBackendCommand = "some_command"
+	defer resetPackageVars()
+
+	_, err := getExecutablePermissions()
+	assert.Error(t, err, "getExecutablePermissions should fail when secretBackendCommand file does not exists")
+}
+
+func setupSecretCommmand(t *testing.T) (string, string) {
+	dir := t.TempDir()
+	t.Cleanup(resetPackageVars)
+
+	secretBackendCommand = filepath.Join(dir, "executable")
+	f, err := os.Create(secretBackendCommand)
+	require.NoError(t, err)
+	f.Close()
+	os.Chmod(secretBackendCommand, 0700)
```

Comment:
No since `os.WriteFile` use umask when creating a file for the first time. This produce different set of rights on windows/OSX/linux. To bypass this we first create the file and then update the permissions.